x-pack/plugins/security_solution/server/endpoint/endpoint_app_context_services.ts

```diff
@@ -276,34 +276,57 @@ export class EndpointAppContextService {
   public getInternalResponseActionsClient({
     agentType = 'endpoint',
     username = 'elastic',
+    taskId,
+    taskType,
   }: {
     agentType?: ResponseActionAgentType;
     username?: string;
+    /** Used with background task and needed for `UnsecuredActionsClient`  */
+    taskId?: string;
+    /** Used with background task and needed for `UnsecuredActionsClient`  */
+    taskType?: string;
   }): ResponseActionsClient {
     if (!this.startDependencies?.esClient) {
       throw new EndpointAppContentServicesNotStartedError();
     }
 
-    if (agentType !== `endpoint`) {
-      throw new EndpointError(
-        `Agent type [${agentType}] does not support usage of response actions via non-HTTP requests!`
-      );
+    let connectorActionsClient =
+      this.startDependencies.connectorActions.getUnsecuredActionsClient();
+
+    // If we have a task id and type, then call is coming from a background task and we need to use those
+    // values with the Action's plugin `UnsecuredActionsClient`'s `.execute()` method. To do so in a
+    // transparent way to the existing response action client, we create a Proxy here and trap the
+    // `GET execute` property and wrap it a function that will automatically inject this data into
+    // `execute()` calls
+    if (taskId && taskType) {
+      connectorActionsClient = new Proxy(connectorActionsClient, {
```

Comment:
why do we need this metaprogramming here? I guess it can be useful in some cases, e.g. if we want to add logging to all methods of a class instead of modifying all of its methods one by one, but here we're just adding an additional layer of complexity to be able to modify one function, if I understood correctly.

could you consider passing the background task related parameters to the function `processPendingActions()` instead? or maybe adding those parameters to the `ConstructorOptions` of `getResponseActionsClient()`, if we want to store those params for the lifetime of the responseActionsClient object?

Reply:
I did not want to introduce Task specific details all the way down to the Action Response client - I feel that would just add unnecessary complexity. The data is only needed to silence a `.warn()`'ing that this new Connector's sub-actions client's `.execute()` method outputs when this data is not set. So its needed for all calls to `.execute()` from the Connector's sub-actions framework - thus I wanted to keep it in a place that would apply only to the use of this client (note: the sub-actions client that is used from our HTTP apis for response actions uses a different Connector client that does not have this need). 

The use of the Connector sub-actions is an internal details to the implementation of the Response Actions Client's, thus I also did not want to expose connector specific details on our class methods and rather keep them as "pure" as possible as far as input arguments go. This will also makes it cleaner should we ever want to make internal changes to how we communicate with external systems by not having to alter our exposed interfaces and associated method signatures. 

I also considered changing the constructor arguments to instead accept on input an instance of the new Class I created (`NormalizedExternalConnectorClient`), but because that class needs the `agentType` I felt it would be best for it to continue to be used only from within the ResponseActionsClient sub-classes where the `agentType` is already know.

Let me know what you think. I can still evaluate to see if there is a more clean and transparent way to do this without having to expose this data down/up the chain of the code.

Reply:
How about this: I'll make some changes to the `NormalizedExternalConnector` class and will add these parameters to it instead and will then use that class instance to initialize the Response Actions Client's. 

Sound good to you?

Reply:
sure, go for it! don't want to hold up the work, so follow up PR is perfect

yeah, using the `NormalizedExternalConnector` sounds reasonable, let's see 👍